distributed/protocol/serialize.py

```diff
@@ -159,6 +160,54 @@ def __init__(self, header, frames):
         self.frames = frames
 
 
+def container_copy(c):
+    if isinstance(c, list):
```

Comment:
I might turn these into explicit type checks (`if type(c) is list` as done in `_extract_serialize`) for a slight speedup. Up to you. Otherwise, this looks good to me.